src/Cpu6502.kt

```diff
@@ -0,0 +1,783 @@
+            Instruction(this::BNE, this::REL, 2), Instruction(this::CMP, this::IZY, 5), Instruction(this::XXX, this::IMP, 2), Instruction(this::XXX, this::IMP, 8), Instruction(this::NOP, this::IMP, 4), Instruction(this::CMP, this::ZPX, 4), Instruction(this::DEC, this::ZPX, 6), Instruction(this::XXX, this::IMP, 6), Instruction(this::CLD, this::IMP, 2), Instruction(this::CMP, this::ABY, 4), Instruction(this::NOP, this::IMP, 2), Instruction(this::XXX, this::IMP, 7), Instruction(this::CPX, this::ABS, 4), Instruction(this::CMP, this::ABX, 4), Instruction(this::DEC, this::ABX, 7), Instruction(this::XXX, this::IMP, 7),
+            Instruction(this::CPX, this::IMM, 2), Instruction(this::SBC, this::IZX, 6), Instruction(this::NOP, this::IMP, 2), Instruction(this::XXX, this::IMP, 8), Instruction(this::CPX, this::ZPG, 3), Instruction(this::SBC, this::ZPG, 3), Instruction(this::INC, this::ZPG, 5), Instruction(this::XXX, this::IMP, 5), Instruction(this::INX, this::IMP, 2), Instruction(this::SBC, this::IMM, 2), Instruction(this::NOP, this::IMP, 2), Instruction(this::XXX, this::IMP, 2), Instruction(this::CPX, this::ABS, 4), Instruction(this::SBC, this::ABS, 4), Instruction(this::INC, this::ABS, 6), Instruction(this::XXX, this::IMP, 6),
+            Instruction(this::BEQ, this::REL, 2), Instruction(this::SBC, this::IZY, 5), Instruction(this::XXX, this::IMP, 2), Instruction(this::XXX, this::IMP, 8), Instruction(this::NOP, this::IMP, 4), Instruction(this::SBC, this::ZPX, 4), Instruction(this::INC, this::ZPX, 6), Instruction(this::XXX, this::IMP, 6), Instruction(this::SED, this::IMP, 2), Instruction(this::SBC, this::ABY, 4), Instruction(this::NOP, this::IMP, 2), Instruction(this::XXX, this::IMP, 7), Instruction(this::NOP, this::IMP, 4), Instruction(this::SBC, this::ABX, 4), Instruction(this::INC, this::ABX, 7), Instruction(this::XXX, this::IMP, 7),
+            )
+    }
+    private fun Fetch(): UByte {
+        if (lookup[opCode].addrmode !== this::IMP) {
+            fetched = Read(addressAbs)
+        }
+        return fetched
+    }
+
+    fun Clock(){
+        if (cycles == 0)
+        {
+            opCode = Read(pc).toInt()
+            SetFlag(Flags.U, true)
+            pc++
+            cycles = lookup[opCode].cycles
+            var additionalCycle1 = (lookup[opCode].addrmode)()
+
+            // Perform operation
+            var additionalCycle2 = (lookup[opCode].opcode)()
+
+            // The addressmode and opcode may have altered the number
+            // of cycles this instruction requires before its completed
+            cycles += (additionalCycle1.toInt() and additionalCycle2.toInt())
+
+            // Always set the unused status flag bit to 1
+            SetFlag(Flags.U, true)
+        }
+
+        cycles -= 1
+    }
```

Comment:
Do we want to have clock cycles implemented?

Reply:
we can disable it for now and come back to it later if we feel like it, tho for each instruction clock method needs to be called to load from memory and so on

Reply:
disbaled it for now